contract/src/internal.rs

```diff
@@ -0,0 +1,322 @@
+        self.set_pool(&token1, p1);
+        self.set_pool(&token2, p1);
+        nep21::ext_nep21::transfer_from(
+            buyer,
+            env::current_account_id(),
+            token1_in.into(),
+            token1,
+            0,
+            SINGLE_CALL_GAS,
+        )
+        .and(nep21::ext_nep21::transfer(
+            recipient,
+            token2_out.into(),
+            token2,
+            0,
+            SINGLE_CALL_GAS,
+        ));
+    }
+
+    pub(crate) fn _price_swap_tokens_in(
+        &self,
+        p_in: &Pool,
+        p_out: &Pool,
+        tokens_in: Balance,
+    ) -> (Balance, Balance) {
+        let near_swap = self.calc_out_amount(tokens_in, p_in.token_bal, p_in.near_bal);
+        let tokens2_out = self.calc_out_amount(near_swap, p_out.near_bal, p_out.token_bal);
+        return (near_swap, tokens2_out);
+    }
+
+    pub(crate) fn _price_swap_tokens_out(
+        &self,
+        p_in: &Pool,
+        p_out: &Pool,
+        tokens_out: Balance,
+    ) -> (Balance, Balance) {
+        let near_swap = self.calc_in_amount(tokens_out, p_out.token_bal, p_out.near_bal);
+        let tokens1_to_pay = self.calc_in_amount(near_swap, p_in.near_bal, p_in.token_bal);
+        return (near_swap, tokens1_to_pay);
+    }
+
+    pub(crate) fn _swap_tokens_exact_in(
+        &mut self,
+        token1: &AccountId,
+        token2: &AccountId,
+        tokens1_paid: Balance,
+        min_tokens2: Balance,
+        buyer: AccountId,
+        recipient: AccountId,
+    ) {
+        assert!(tokens1_paid > 0 && min_tokens2 > 0, "E2");
+        assert_ne!(token1, token2, "E9");
+        let mut p1 = self.get_pool(&token1);
+        let mut p2 = self.get_pool(&token2);
+        let (near_swap, tokens2_out) = self._price_swap_tokens_in(&p1, &p2, tokens1_paid);
+        assert!(tokens2_out >= min_tokens2, "E7");
+
+        self._swap_tokens(
+            &mut p1,
+            &mut p2,
+            token1,
+            token2,
+            tokens1_paid,
+            tokens2_out,
+            near_swap,
+            buyer,
+            recipient,
+        )
+    }
+
+    pub(crate) fn _swap_tokens_exact_out(
+        &mut self,
+        token1: &AccountId,
+        token2: &AccountId,
+        tokens2_out: Balance,
+        max_tokens1_paid: Balance,
+        buyer: AccountId,
+        recipient: AccountId,
+    ) {
+        assert!(tokens2_out > 0 && max_tokens1_paid > 0, "E2");
+        assert_ne!(token1, token2, "E9");
+        let mut p1 = self.get_pool(&token1);
+        let mut p2 = self.get_pool(&token2);
+        let (near_swap, tokens1_to_pay) = self._price_swap_tokens_out(&p1, &p2, tokens2_out);
+        assert!(tokens1_to_pay >= max_tokens1_paid, "E8");
+
+        self._swap_tokens(
+            &mut p1,
+            &mut p2,
+            token1,
+            token2,
+            tokens1_to_pay,
+            tokens2_out,
+            near_swap,
+            buyer,
+            recipient,
+        )
+    }
+}
```

Comment:
Problem with moving content to a new file is that in github PR we don't see if there are any changes.

I see that you changed method privacy to `pub(crate)`. Are there any other changes?